src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,59 @@
-export const PeopleFilters = () => {
+import { useCallback, useEffect, useState } from 'react';
+import { SearchLink } from './SearchLink';
+import classNames from 'classnames';
+import { useSearchParams } from 'react-router-dom';
+
+export const PeopleFilters: React.FC = () => {
+  const [nameFilter, setNameFilter] = useState<string>('');
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const sex = searchParams.get('sex');
+  const centuries = searchParams.getAll('centuries');
+
+  // If the query is empty, remove it from the URL
+  useEffect(() => {
+    if (!nameFilter.length) {
+      searchParams.delete('query');
+      setSearchParams(searchParams);
+    } else {
+      searchParams.set('query', nameFilter);
+      setSearchParams(searchParams);
+    }
+  }, [nameFilter, searchParams, setSearchParams]);
+
+  const getCenturies = useCallback(
```

Comment:
No need to memoize it, there is no need to think about referential equality or any complex computations. 
As the discussion point: you are passing an array as the dependency. Will it always work as expected? :)